transport/pool_integration_test.go

```diff
@@ -51,3 +54,59 @@ func TestNodeConnPoolIntegration(t *testing.T) {
 		}
 	}
 }
+
+func TestConnPoolGetConn(t *testing.T) {
```

Comment:
Conn not GetConn

Reply:
I think this test is kind of a smoke test and it seems it to me it's not correct.
From the fact that a connection is returned you cannot draw a conclusion that its Waiting value changes.
Please remove the test.

the gist of it can be added to the test that exists.

Also i

Reply:
> From the fact that a connection is returned you cannot draw a conclusion that its Waiting value changes.

I agree that's why after drawing connection I call `_, _ = conn.Supported()`.
Which calls `c.sendRequest` and it calls `c.r.setHandler(h)` which although may return error and won't change connection Waiting value (it's returning error when `connReader.closed == true` or `c.s.Alloc()` fails when all stream IDs are busy), but in the span of having semi-new connection and 100k requests I think extremely unlikely that it will fail.
If `c.r.setHandler(h)` doesn't fail then connection Waiting value changes.

If I have a flaw in my thought process please correct me and I will incorporate the MurmurToken part to the `TestNodeConnPoolIntegration` and remove the rest.